### PR TITLE
Create CMakeLists.txt #57

### PR DESCRIPTION
**Pull request by Barto-Paja**
RE: issue #57
https://github.com/ftylitak/qzxing/issues/57

Tested on: Qt 5.9.5 CMake 3.2 Win7 64bit

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+find_package(Qt5 REQUIRED Core)
+find_package(Qt5 REQUIRED Network)
+find_package(Qt5 REQUIRED Xml)
+find_package(Qt5 REQUIRED Multimedia)
+find_package(Qt5 REQUIRED MultimediaWidgets Widgets)
+find_package(Qt5 REQUIRED Svg Quick QuickControls2)
+
+set(CMAKE_AUTOMOC ON)
+#set(CMAKE_BUILD_TYPE Debug)
+add_subdirectory(zxing)
```
Comment (simonschmeisser) [inline src/CMakeLists.txt]: you did not include a zxing/CMakeLists.txt, did you miss it?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+    QZXing.h
+    QZXingFilter.cpp
+    QZXingFilter.h
+    QZXingImageProvider.cpp
+    QZXingImageProvider.h
+    QZXing_global.h
+
+# zxing/bigint
+
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/bigint/BigInteger.hh
```
Comment (simonschmeisser) [inline src/CMakeLists.txt]: do you really need `${CMAKE_CURRENT_SOURCE_DIR} ` here (and on all following lines?)

**Comment by simonschmeisser** [comment]
Having a proper CMake file for this project would be great!

File: src/CMakeLists.txt (hunk 1 above)
Comment (Barto-Paja) [reply src/CMakeLists.txt]: It was my mistake. zxing/CMakeLists.txt Was not necessary.

File: src/CMakeLists.txt (hunk 2 above)
Comment (Barto-Paja) [reply src/CMakeLists.txt]: I wrote this as a beginner, so all internal CMakeLists have been removed, so $ {CMAKE_CURRENT_SOURCE_DIR} is required.

**Comment by Barto-Paja** [comment]
I fixed my mistake https://github.com/Barto-Paja/qzxing/commit/85945f9af6350d5be7e55db02673ff3115a3aa8d

Sorry for being late.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,323 @@
+cmake_minimum_required(VERSION 3.2)
+project(QZXing)
+
+find_package(Qt5 REQUIRED Core)
+find_package(Qt5 REQUIRED Network)
+find_package(Qt5 REQUIRED Xml)
+find_package(Qt5 REQUIRED Multimedia)
+find_package(Qt5 REQUIRED MultimediaWidgets Widgets)
+find_package(Qt5 REQUIRED Svg Quick QuickControls2)
```
Comment (Talkless) [inline src/CMakeLists.txt]: Usage of Quick and Multimedia should be configurable, as it is currently with `CONFIG += qzxing_qml` and `CONFIG += qzxing_multimedia`

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    Qt5::MultimediaWidgets )
+
+file(GLOB PUBLIC_HEADER QZXing.h)
+
+# Change Global Definitions depending on how you want to use the library
+target_compile_definitions(qzxing
+  PRIVATE -QZXING_LIBRARY
+  PRIVATE -ZXING_ICONV_CONST
+  PUBLIC  -QZXING_QML
+  PUBLIC  -QZXING_MULTIMEDIA
```
Comment (Talkless) [inline src/CMakeLists.txt]: Same, Qml amd Multimedia should be configurable, enabled on demand..?

Comment (Barto-Paja) [reply src/CMakeLists.txt]: I added new parameters:
-QZXING_MULTIMEDIA set ON for use  QZXingFilter
-QZXING_USE_QML set ON for use QZXING + QML

you can try now there: 8eefa1d653d29729a8336fa8d09284e23b50bf2e
When I do more settings from pri file, I'll send next pull request.

**Pull request merged**